tools/shared/pragma.c

```diff
@@ -430,6 +431,7 @@ struct c {
 static struct c table[] = {
     {"align", SW_ALIGN, false, S_NONE, S_NONE},
     {"altcode", SW_ALTCODE, true, S_LOOP, S_LOOP | S_ROUTINE | S_GLOBAL},
+    {"alwaysinline", SW_ALWAYSINLINE, true, S_ROUTINE, S_ROUTINE | S_GLOBAL},
```

Comment:
I recommend naming the directive `INLINE` or `FORCEINLINE` instead of `ALWAYSINLINE`, since they seem to be used by other Fortran compilers (Intel, Cray). If you go with `INLINE`, you can also leverage Flang's support for the optional `NO` prefix to consolidate the processing of the directives.

Reply:
Well, `NOINLINE` already exists. Only force/alwaysinline is required. I went with `alwaysinline` because that's the attribute in clang/llvm. That said, I am changing `alwaysinline` to `forceinline` for, as you said, compatibility reasons.